### DISCUSSION TITLE
To consider the displaced floating platform position to derive the wave-excitation terms in HydroDyn

### DISCUSSION THREAD

**Discussion by amoratoc** [OpenFAST/openfast#875] [Ideas]
All of the wave-excitation terms in HydroDyn, including 2nd order forces, are derived using wave kinematics at the undisplaced position of the floating platform. While this assumption does not have important effects on offshore fixed structures, it may play a relevant role in some floating solutions, specially when using single point mooring technology. Such hydrodynamic loads may act as a restoring forces in sway-yaw.

Thank you!
Àlex

**Comment by andrew-platt** [comment]
Hi Àlex,

This is actually a feature that is currently in development.  We expect to release this within a few months.

@jjonkman, can you comment further on the timeline?

Andy

**Comment by jjonkman** [comment]
I'll clarify that the inclusion of platform displacement in the wave-excitation terms of the OpenFAST HydroDyn module is in a project that was recently initiated, but we have not actually started the development.  The project (ARPA-E ATLANTIS WEIS) is first focusing on defining the functional requirements of a new tool set to enable controls co-design of floating offshore wind turbines, including the development path and time frame of various upgrades to OpenFAST.  This will then get reviewed by an external advisory board before being finalized.  So, it is difficult to say at this time what the prioritization of this specific OpenFAST upgrades will be.

Regardless, I would expect that we'd work on the inclusion of platform displacement in the wave-excitation terms and complete it within the 2-year project.  It should be clarified that this specific task is not entirely straightforward because 2nd-order terms should not be duplicated in the implementation.

**Comment by amoratoc** [comment]
Many thanks Andrew and Jason for the clarification.
Àlex
